src/drivers/imx/dummy-dma.c

```diff
@@ -45,54 +134,302 @@ static int dummy_dma_status(struct dma_chan_data *channel,
 			    struct dma_chan_status *status,
 			    uint8_t direction)
 {
+	struct dma_chan_pdata *ch = dma_chan_get_data(channel);
+
+	(void)direction; /* TODO should we check it? */
+
+	status->state = channel->status;
+	status->flags = 0; /* TODO What flags should be put here? */
+	status->r_pos = ch->r_pos;
+	status->w_pos = ch->w_pos;
+
+	status->timestamp = 0; /* TODO what would be the correct one? */
 	return 0;
 }
 
 /* set the DMA channel configuration, source/target address, buffer sizes */
 static int dummy_dma_set_config(struct dma_chan_data *channel,
 				struct dma_sg_config *config)
 {
-	return 0;
+	struct dma_chan_pdata *ch = dma_chan_get_data(channel);
+	uint32_t flags;
+	int ret = 0;
+
+	spin_lock_irq(channel->dma->lock, flags);
+
+	if (!config->elem_array.count) {
+		trace_dummydma_error("dummy-dmac: %d channel %d no DMA descriptors",
+				     channel->dma->plat_data.id,
+				     channel->index);
+
+		ret = -EINVAL;
+		goto out;
+	}
+
+	channel->direction = config->direction;
+	/* TODO Should we keep this check? */
+	if (config->direction != DMA_DIR_HMEM_TO_LMEM &&
```

Comment:
Yes, I think we should keep this check. Dummy-dma is only used (now) for transferring data from Host memory to Local memory (and vice versa).

Reply:
I think it can be changed to an assertion though, given that normally if no other directions are allowed in the DMA controller structures in the platform/lib/dma.c file they shouldn't be requested. Will remove the TODO for now.